addons/mail/static/src/models/thread_view/thread_view.js

```diff
@@ -116,10 +116,14 @@ function factory(dependencies) {
         _computeThreadShouldBeSetAsSeen() {
             // FIXME condition should not be on "composer is focused" but "threadView is active"
             // See task-2277543
-            const lastMessageIsVisible = this.lastVisibleMessage &&
-                this.lastVisibleMessage === this.lastMessage;
-            if (lastMessageIsVisible && this.hasComposerFocus && this.thread) {
-                this.thread.markAsSeen(this.lastMessage.id).catch(e => {
+            if (
+                this.lastVisibleMessage &&
+                this.lastNonTransientMessage &&
+                this.lastVisibleMessage.id >= this.lastNonTransientMessage.id &&
```

Comment:
I think here is the part of the code we discussed would benefit from having a `lastNonTransientVisibleMessage` so that "lastMessageIsVisible" is just `this.lastNonTransientVisibleMessage === this.lastNonTransientMessage`